benchmark/python/benchmark_e2e.py

```diff
@@ -0,0 +1,184 @@
+    avg_per_token_prompt_thrpt = batch_size * (1 / (avg_per_token_prompt_latency_ms / 1000))
+    print(f"Average Prompt Processing Latency (per token): {avg_per_token_prompt_latency_ms} ms")
+    print(f"Average Prompt Processing Throughput (per token): {avg_per_token_prompt_thrpt} tps")
+
+    # Calculate token generation input prep metrics
+    avg_token_gen_latency_s = sum(token_gen_times) / len(token_gen_times)
+    avg_token_gen_latency_ms = avg_token_gen_latency_s * 1000
+    avg_token_gen_thrpt = batch_size * (1 / avg_token_gen_latency_s)
+    print(f"Average Token Generation Latency (per token): {avg_token_gen_latency_ms} ms")
+    print(f"Average Token Generation Throughput (per token): {avg_token_gen_thrpt} tps")
+    
+    # Calculate sampling metrics
+    avg_sampling_latency_s = sum(sampling_times) / len(sampling_times)
+    avg_sampling_latency_ms = avg_sampling_latency_s * 1000
+    avg_sampling_thrpt = batch_size * (1 / avg_sampling_latency_s)
+    print(f"Average Sampling Latency (per token): {avg_sampling_latency_ms} ms")
+    print(f"Average Sampling Throughput (per token): {avg_sampling_thrpt} tps")
+
+    all_csv_metrics = [[
+        batch_size, 
+        prompt_length, 
+        avg_tokenization_thrpt, 
+        avg_tokenization_latency_ms, 
+        avg_per_token_prompt_thrpt, 
+        avg_per_token_prompt_latency_ms, 
+        avg_token_gen_thrpt, 
+        avg_token_gen_latency_ms, 
+        avg_sampling_thrpt, 
+        avg_sampling_latency_ms,
+    ]]
+
+    # Add metrics to CSV
+    if args.verbose: print("Adding results to CSV")
+    filename = args.output + ".csv"
+    save_results(all_csv_metrics, filename)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="End-to-end benchmarking for gen-ai")
+    parser.add_argument('-i', '--input_folder', type=str, required=True, help='Onnx model folder path (must contain config.json and model.onnx)')
+    parser.add_argument('-b', '--batch_size', type=int, default=1, help='Number of sequences to generate in parallel')
+    parser.add_argument('-l', '--prompt_length', type=int, default=16, help='Number of tokens for prompt')
+    parser.add_argument('-g', '--generation_length', type=int, default=256, help='Number of tokens to generate after prompt')
+    parser.add_argument('-r', '--repetitions', type=int, default=10, help='Number of times to repeat the benchmark')
+    parser.add_argument('-w', '--warmup', type=int, default=5, help='Number of warmup runs before benchmarking')
+    parser.add_argument('-k', '--top_k', type=int, default=50, help='Top k tokens to sample from')
+    parser.add_argument('-p', '--top_p', type=float, default=1.0, help='Top p probability to sample with')
+    parser.add_argument('-o', '--output', type=str, default='genai_e2e', help='Output CSV file name or path')
+    parser.add_argument('-d', '--device', type=str, choices=['cpu', 'cuda'], default='cpu', help='Device to use, default is CPU, use CUDA for GPU')
```

Comment:
Since this script is only benchmarking ONNX Runtime GenAI, should we change `-d/--device` to `-ep/--execution_provider`?